presto-docs/src/main/sphinx/admin/properties.rst

```diff
@@ -58,6 +60,251 @@ General Properties
+    It may be required to increase this value if data are skewed.
+
+
+.. _query-max-run-time:
+
+``query.max-run-time``
+^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``duration``
+    * **Default value:** ``100d``
+
+    Used as default for session property ``query_max_run_time``. Queries that last longer than the duration set with
+    ``query.max-run-time`` will be cancelled. It is important to set this value to much
+    higher than the average query time to avoid problems with outliers (some queries may randomly take much longer due
+    to cluster load and other circumstances). If Presto runs in an environment where there are mostly very short queries
+    then it may be a good idea to set it to a small value to detect user errors in queries.
+    It may also be decreased in poor Presto cluster configuration with mostly short queries to increase garbage
+    collection efficiency and by that lowering memory usage in cluster. As the query timed out by this limit immediately
+    returns all used resources this may be particularly useful in query management systems to force user limits. The
+    session property ``query_max_run_time`` may also be set to a value less than ``query.max-run-time`` in order to
+    crosscheck for bugs in the query.
+
+
+``query.max-cpu-time``
+^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``duration``
+    * **Minimum Value:** ``1ns``
+    * **Default value:** ``1000000000d``
+
+    Used as default for session property ``query_max_cpu_time``. Similar to ``query.max-run-time`` but this sets a
+    maximum CPU time limit of a query. Queries whose CPU time hits a value more than the duration set with
+    ``query.max-cpu-time`` will be cancelled.
+
+
+``query.max-execution-time``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``duration``
+    * **Default value:** ``100d``
+
+    Used as default for session property ``query_max_execution_time``. Unlike ``query.max-run-time`` which enforces
+    starting from the query creation time, this property does not consider the queueing time.
+
+
+``query.queue-config-file``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Default value:**
+
+    **Deprecated** The path to the queue config file. Queues are used to manage the number of concurrent queries across
+    the system. More information on queues and how to configure them can be found in :doc:/admin/queue.
+
+
+``query.remote-task.max-callback-threads``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``integer``
+    * **Minimum Value:** ``1``
+    * **Default value:** ``1000``
+
+    This value describes the maximum size of the thread pool used to handle responses to HTTP requests for
+    each task. Increasing this value will cause more resources to be used for handling HTTP communication
+    itself, but may also improve response time when Presto is distributed across many hosts or there are
+    a lot of small queries being run.
```

Comment:
>> a lot of small queries being run.

It would be nice to rephrase this. First, `small queries` is a little bit vague here, I think you mean in terms of the number of tasks it creates, so maybe we should rephrase this a bit to make it explicit. 

Also the end of sentence `being run` sounds weird.